pmlc/conversion/gpu_to_comp/gpu_to_comp.cc

```diff
@@ -80,8 +80,32 @@ mlir::LogicalResult
 RewriteLaunchFunc::matchAndRewrite(gpu::LaunchFuncOp op,
                                    mlir::PatternRewriter &rewriter) const {
   mlir::Location loc = op.getLoc();
+  // Look up the containing function.
+  auto func = op.getParentOfType<mlir::FuncOp>();
+  if (!func) {
+    return mlir::failure();
+  }
+  // Add the execution device parameter to the containing function, unless it
+  // already has one.
+  auto funcTy = func.getType();
+  mlir::Value device;
+  if (funcTy.getNumInputs() && funcTy.getInput(0).isa<comp::DeviceType>()) {
```

Comment:
I don't like the idea that we are saying the the first argument to the function has to be a device. Let's chat about different design choices here.